external-crates/move/move-compiler/src/shared/mod.rs

```diff
@@ -404,6 +411,9 @@ impl CompilationEnv {
                     );
                     diag.add_note(help)
                 }
+                if self.flags().warnings_are_errors() {
+                    diag = diag.set_severity(Severity::NonblockingError)
+                }
```

Comment:
It would take some retooling, but I wonder if we could reworking warning filters so we could include warnings-as-errors as a warning filter. I'm picturing something like an impl for warning filters as:

```
let filter = self.warning_filter.last();
if let Some(mut diag) = filter.filter_diag(diag) {
  ...
}
```

Then in the impl block on warning filters:

```
fn filter_diag(&self, mut diag: Diagnostic) -> Option<Diagnostic> {
  match self {
    ...
     WarningFilter::All => None,
     WarningFilter::ErrorsAsWarnings => {
       Some(diag.set_severity(Severity::NonBlockingError))
     } 
   }
 }
```

Reply:
Interesting... whats the goal here? Because today we will still honor the `#[allow(unused)]` annotations, before lifting them to errors. Which seems desirable/nice

Reply:
Presumably those annotations would also be part of this chain (nesting would recur and hand things around or similar). The goal is just to unify all of the logic in one place.